src/providers/wms/qgswmsprovider.cpp

```diff
@@ -615,11 +615,16 @@ static void _drawDebugRect( QPainter& p, const QRectF& rect, const QColor& color
 
 QImage *QgsWmsProvider::draw( QgsRectangle const & viewExtent, int pixelWidth, int pixelHeight, QgsRasterBlockFeedback* feedback )
 {
-  QgsDebugMsg( "Entering." );
+  QImage* image = new QImage( pixelWidth, pixelHeight, QImage::Format_ARGB32 );
```

Comment:
make sense to use standard QgsLayerItem::iconRaster() as in:
https://qgis.org/api/2.18/qgslayertreemodel_8cpp_source.html#l00234
so we maintain the standard raster icon

Reply:
Probably not directly jere because this is a general draw method, not explicitely for preview images. I guess this needs to be done on a different level.

Reply:
you are right... I'll prepare to set default iconRaster in https://qgis.org/api/2.18/qgslayertreemodel_8cpp_source.html#l00234

Reply:
Why do we return pointers here anyway? QImage is implicitly shared.

Reply:
Because it's 2.18 and api frozen :P

Reply:
Last I checked private methods aren't considered frozen api ;)

Reply:
There are also nullptr checks on returned values. I think I'll stick to making this change not too intrusive still ;)